source/compiler/qsc_eval/src/intrinsic/tests.rs

```diff
@@ -1910,6 +1910,871 @@ fn check_pauli_noise() {
     );
 }
 
+#[test]
+fn qubit_loss_generates_loss_result() {
+    check_intrinsic_result(
+        "",
+        indoc! {"{
+            use q = Qubit();
+            Std.Diagnostics.ConfigureQubitLoss(1.0);
+            MResetZ(q)
+        }"},
+        &expect!["Loss"],
+    );
+}
+
+#[test]
+fn checked_measurement_with_normal_qubit_returns_false() {
+    check_intrinsic_result(
+        "",
+        indoc! {"{
+            use q = Qubit();
+            MResetZChecked(q)
+        }"},
+        &expect!["(Zero, false)"],
+    );
+}
+
+#[test]
+fn checked_measurement_with_lost_qubit_returns_true() {
+    check_intrinsic_result(
+        "",
+        indoc! {"{
+            use q = Qubit();
+            Std.Diagnostics.ConfigureQubitLoss(1.0);
+            MResetZChecked(q)
+        }"},
+        &expect!["(Loss, true)"],
+    );
+}
+
+#[test]
+fn comparison_of_loss_result_runtime_error() {
+    check_intrinsic_result(
+        "",
+        indoc! {"{
+            use q = Qubit();
+            Std.Diagnostics.ConfigureQubitLoss(1.0);
+            MResetZ(q) == One
```

Comment:
Can you add another test that compares loss with itself?

Reply:
That's a good idea, I'll include a test to confirm that behavior!